blog/2019-12-18-HowToPlanAndOptimizeASecretSanta.adoc

```diff
@@ -0,0 +1,102 @@
+= How to Plan (and Optimize) a Secret Santa
+:page-interpolate: true
+:awestruct-author: cchianel
+:awestruct-layout: blogPostBase
+:awestruct-tags: [useCase]
+
+Many businesses host a "Secret Santa", where each employee get assigned
+a coworker whom they need to buy a present for. This fosters good relations
+between the employees and bring them closer together. But what about global
+companies, where coworkers may be many miles apart? Ideally, we want
+employees who are further apart to give gifts to each other, since they
+are the ones who probably know the least about each other. Let optimize
+it with OptaPlanner!
+
+== The Constraints
+
+One of the most obvious constraint is that everyone should get a gift.
+Or phrased a little differently: no one should get multiple gifts.
+
+Here is how one would create this constraint in OptaPlanner using the
+constraint streams API:
+[source,java]
+----
+private Constraint sameRecieverConflict(ConstraintFactory constraintFactory) {
+    return constraintFactory
+            .fromUniquePair(SecretSantaAssignment.class,
+                    Joiners.equal(SecretSantaAssignment::getReciever))
+            .penalize("Same Reciever", HardMediumSoftBigDecimalScore.ONE_HARD);
+}
+----
+
+Next, we need to to award assignments where the gifter and the reciever
+are farther apart. Here how we will do it with constraint streams:
+[source,java]
+----
+private Constraint largerDistanceAward(ConstraintFactory constraintFactory) {
+    return constraintFactory
+            .from(SecretSantaAssignment.class)
+            .rewardConfigurableBigDecimal("secretFactor", "Larger Distance Award",
+                              (m) -> BigDecimal.valueOf(Location.calculateDistanceBetween(m.getGifter().getLocation(),
+                                                                       m.getReciever().getLocation())));
+}
+----
+
+Let try solving with just these two constraints to see what we get:
+
+image::noPairChecking.png[]
+
+Alice is assigned to Daniella, Daniella is assigned to Alice...
+Bob is assigned to Austin, Austin is assigned to Bob...
+Something about this solution seems odd; it only generate assignments
+that give gifts to one another!
+
+== What happened?
+
+This is not odd at all, but rather a consequence of how we defined
+our constraints: if Alice the person furthest from Daniella, then
+Daniella is most likely the person furthest from Alice. This means our
+"Larger Distance Award" constraint will strongly favour pairs of employees.
```

Comment:
"favor" instead of "favour"?

Reply:
Depends on if you user American vs European spelling. Favour is the variant used in Canada and Europe, Favor is used in America. Is American or European spelling preferred for blog posts?

Reply:
True. I assumed American english but guess that's a @ge0ffrey question :)

Reply:
In my opinion, blog posts are supposed to be personal and therefore don't need to adhere to some strict guidelines.
That said, when we write docs, we stick to American spelling.